// @highlight-run/react/dist/index.js
"use strict";var K=Object.create;var m=Object.defineProperty,Q=Object.defineProperties,X=Object.getOwnPropertyDescriptor,oo=Object.getOwnPropertyDescriptors,eo=Object.getOwnPropertyNames,H=Object.getOwnPropertySymbols,to=Object.getPrototypeOf,z=Object.prototype.hasOwnProperty,ro=Object.prototype.propertyIsEnumerable;var A=(t,e,o)=>e in t?m(t,e,{enumerable:!0,configurable:!0,writable:!0,value:o}):t[e]=o,g=(t,e)=>{for(var o in e||(e={}))z.call(e,o)&&A(t,o,e[o]);if(H)for(var o of H(e))ro.call(e,o)&&A(t,o,e[o]);return t},h=(t,e)=>Q(t,oo(e));var no=(t,e)=>()=>(e||t((e={exports:{}}).exports,e),e.exports),io=(t,e)=>{for(var o in e)m(t,o,{get:e[o],enumerable:!0})},O=(t,e,o,i)=>{if(e&&typeof e=="object"||typeof e=="function")for(let n of eo(e))!z.call(t,n)&&n!==o&&m(t,n,{get:()=>e[n],enumerable:!(i=X(e,n))||i.enumerable});return t};var v=(t,e,o)=>(o=t!=null?K(to(t)):{},O(e||!t||!t.__esModule?m(o,"default",{value:t,enumerable:!0}):o,t)),ao=t=>O(m({},"__esModule",{value:!0}),t);var R=no((mo,w)=>{"use strict";(function(){"use strict";var t={}.hasOwnProperty;function e(){for(var n="",a=0;a<arguments.length;a++){var s=arguments[a];s&&(n=i(n,o(s)))}return n}function o(n){if(typeof n=="string"||typeof n=="number")return n;if(typeof n!="object")return"";if(Array.isArray(n))return e.apply(null,n);if(n.toString!==Object.prototype.toString&&!n.toString.toString().includes("[native code]"))return n.toString();var a="";for(var s in n)t.call(n,s)&&n[s]&&(a=i(a,s));return a}function i(n,a){return a?n?n+" "+a:n+a:n}typeof w!="undefined"&&w.exports?(e.default=e,w.exports=e):typeof define=="function"&&typeof define.amd=="object"&&define.amd?define("classnames",[],function(){return e}):window.classNames=e})()});var uo={};io(uo,{ErrorBoundary:()=>E,ReportDialog:()=>S,SampleBuggyButton:()=>fo});module.exports=ao(uo);var u=v(require("react"));var r=v(require("react")),F=v(R());var M="ba9c647cce52083170869d8ab4be319f402bcaf786e5d7099b013d5757ef5c9d",lo=`._container_18f8u_1 {
	--brand-color: #744ed4;
	--brand-color-dark: #5629c6;
	--color-text-primary: #111111;
	--color-text-primary-inverted: #fff;
	--color-text-link: var(--brand-color);
	--size-large: 24px;
	--size-medium: 16px;
	--size-small: 12px;
	--size-xSmall: 8px;

	--color-white: #fff;
	--color-gray-100: hsla(0, 0%, 93.7%, 0.49019607843137253);
	--color-gray-200: #f2f2f2;
	--color-gray-300: #eaeaea;
	--color-gray-400: #bdbdbd;
	--color-gray-500: #828282;
	--color-gray-600: #7e7e7e;
	--color-gray-700: #8a8f98;
	--color-gray-800: #2d2f36;

	--font-family: Inter, -apple-system, BlinkMacSystemFont, 'Segoe UI',
		Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji',
		'Segoe UI Emoji', 'Segoe UI Symbol';

	background: hsl(0deg 0% 0% / 75%);
	font-family: var(--font-family);
	position: fixed;
	top: 0;
	left: 0;
	width: 100vw;
	height: 100vh;
	z-index: 999999;
	backdrop-filter: blur(1px);
	display: flex;
	justify-content: center;
	align-items: center;
}

@media only screen and (max-width: 420px) {
	._card_18f8u_41 {
		max-width: 100vw !important;
		height: 100%;
		border-radius: 0;
	}

	._cardContents_18f8u_47 {
		height: 100%;
		display: flex;
		flex-direction: column;
	}

	._form_18f8u_53 {
		height: 100%;
	}

	._textareaLabel_18f8u_57 {
		flex: 1;
	}

	._form_18f8u_53 textarea {
		height: 100%;
	}

	._ad_18f8u_65 {
		display: none;
	}
}

._card_18f8u_41 {
	border-radius: var(--size-xSmall);
	padding: 20px;
	width: 420px;
	max-width: calc(100vw - 1rem);
	background: white;
}

._title_18f8u_78 {
	font-size: 20px;
	color: var(--color-text-primary);
}

._subtitle_18f8u_83 {
	font-size: 13px;
	margin-top: var(--size-medium) !important;
	font-weight: 400;
	color: #8c8b92;
}

._form_18f8u_53 {
	padding-top: var(--size-large);
	display: flex;
	flex-direction: column;
	gap: var(--size-small);
	width: 100%;
}

._form_18f8u_53 label {
	display: flex;
	flex-direction: column;
	font-size: 11px !important;
	font-weight: 500;
	color: var(--color-gray-500);
}

._form_18f8u_53 input {
	height: 28px;
}

._form_18f8u_53 textarea {
	font-family: var(--font-family);
	min-height: 80px;
}

._form_18f8u_53 input,
._form_18f8u_53 textarea {
	background-color: var(--color-white);
	border: 1px solid #dcdbdd;
	border-radius: 6px;
	color: var(--color-gray-500);

	font-size: 13px;
	margin: 8px 0;
	outline: none;
	padding: 4px 6px;
	width: calc(100%);
}

._form_18f8u_53 button {
	align-items: center;
	display: flex;
	justify-content: center;
	height: auto;
	color: var(--color-text-primary-inverted);
	background-color: var(--brand-color);
	text-shadow: none;
	box-shadow: none;
	padding: 4px 6px;
	font-size: 14px;
	border-radius: 6px;
	border: 0;
	cursor: pointer;
}

._form_18f8u_53 button:hover {
	background-color: var(--brand-color-dark);
}

._form_18f8u_53 button[disabled] {
	opacity: 0.5;
	pointer-events: none;
}

._form_18f8u_53 ._closeButton_18f8u_154 {
	background-color: transparent;
	color: #6f6e77 !important;
	font-size: 13px;
	font-family: var(--font-family);
	border: 1px solid var(--color-white);
}

._form_18f8u_53 ._closeButton_18f8u_154:hover {
	background-color: var(--color-gray-100);
	border-color: var(--color-gray-300);
}

._confirmationButton_18f8u_167 {
	float: right;
	margin-top: 3rem;
}

._formFooter_18f8u_172 {
	display: flex;
	flex-direction: row-reverse;
	flex-wrap: wrap;
	align-items: center;
	justify-content: space-between;
}

._formActionsContainer_18f8u_180 {
	display: flex;
	flex-direction: row-reverse;
	gap: 6px;
}

._ad_18f8u_65:hover {
	background-color: var(--color-gray-100);
}

._logoContainer_18f8u_190 {
	display: flex;
	align-items: center;
	gap: 6px;

	border: 1px solid var(--color-gray-300);
	border-radius: 6px;
	color: var(--color-gray-500);
	font-size: 13px;
	padding: 4px 8px;
}

._logo_18f8u_190 {
	height: 16px;
}

._buggyButton_18f8u_206 {
	align-items: center;
	background: #5629c6;
	border-radius: 8px;
	border: 1px solid #bdbdbd;
	box-shadow: none;
	color: #fff;
	cursor: pointer;
	display: flex;
	font-family: Inter, -apple-system, BlinkMacSystemFont, 'Segoe UI',
		Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji',
		'Segoe UI Emoji', 'Segoe UI Symbol';
	font-size: 14px;
	height: auto;
	justify-content: center;
	min-height: 40px;
	padding: 4px 16px;
	text-shadow: none;
}
`;(function(){if(typeof document!="undefined"&&!document.getElementById(M)){var t=document.createElement("style");t.id=M,t.textContent=lo,document.head.appendChild(t)}})();var l={container:"_container_18f8u_1",card:"_card_18f8u_41",cardContents:"_cardContents_18f8u_47",form:"_form_18f8u_53",textareaLabel:"_textareaLabel_18f8u_57",ad:"_ad_18f8u_65",title:"_title_18f8u_78",subtitle:"_subtitle_18f8u_83",closeButton:"_closeButton_18f8u_154",confirmationButton:"_confirmationButton_18f8u_167",formFooter:"_formFooter_18f8u_172",formActionsContainer:"_formActionsContainer_18f8u_180",logoContainer:"_logoContainer_18f8u_190",logo:"_logo_18f8u_190",buggyButton:"_buggyButton_18f8u_206"};function S({labelClose:t="Cancel",labelComments:e="Message",labelName:o="Name",labelEmail:i="Email",labelSubmit:n="Submit",subtitle2:a="If you\u2019d like to help, tell us what happened below.",subtitle:s="Our team has been notified.",successMessage:f="Your feedback has been sent. Thank you!",successSubtitle:d="Thank you for sending us feedback. If you have any other concerns/questions, reach out to this application's support email.",title:_="It looks like we\u2019re having issues.",placeholderComments:T="I typed in a name then clicked the button",user:p,onCloseHandler:D,onSubmitHandler:B,hideHighlightBranding:U=!1,error:k}){let[b,L]=(0,r.useState)((p==null?void 0:p.name)||""),[y,P]=(0,r.useState)((p==null?void 0:p.email)||""),[x,V]=(0,r.useState)(""),[Z,N]=(0,r.useState)(!1),[W,Y]=(0,r.useState)(!1),$=(0,r.useRef)(new Date().toISOString()),q=(0,r.useMemo)(()=>{let c=!!y.match(/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i);return!!b&&c&&!!x},[b,y,x]);r.default.useEffect(()=>{var c;(c=window==null?void 0:window.H)!=null&&c.consumeError&&k&&window.H.consumeError(k)},[k]);let J=c=>{var I;c.preventDefault(),N(!0),(I=window==null?void 0:window.H)!=null&&I.addSessionFeedback?window.H.addSessionFeedback({verbatim:x,userName:b,userEmail:y,timestampOverride:$.current}):console.warn("Highlight is not initialized. Make sure highlight.run is installed and running."),new Promise(G=>window.setTimeout(G,300)).then(()=>{N(!1),Y(!0),B&&B()})};return r.default.createElement(r.default.Fragment,null,r.default.createElement("style",{dangerouslySetInnerHTML:{__html:`
					@font-face {
						font-display: swap;
						font-family: 'Inter';
						font-style: normal;
						font-weight: normal;
						src: local('Inter Regular'), local('InterRegular'),
							url('https://app.highlight.io/font/Inter-Regular.woff2')
								format('woff2');
					}
					@font-face {
						font-display: swap;
						font-family: 'Inter';
						font-style: normal;
						font-weight: 500;
						src: local('Inter Medium'), local('InterMedium'),
							url('https://app.highlight.io/font/Inter-Medium.woff2')
								format('woff2');
					}

					::placeholder {
						color: var(--color-gray-300);
					}
			`}}),r.default.createElement("main",{className:l.container},r.default.createElement("div",{className:l.card},W?r.default.createElement("div",{className:l.cardContents},r.default.createElement("h1",{className:l.title},f),r.default.createElement("h4",{className:l.subtitle},d),r.default.createElement("div",null,r.default.createElement("button",{className:(0,F.default)(l.button,l.confirmationButton),onClick:D},"Close"))):r.default.createElement("div",{className:l.cardContents},r.default.createElement("div",null,r.default.createElement("h1",{className:l.title},_),r.default.createElement("h2",{className:l.subtitle},s," ",a)),r.default.createElement("form",{className:l.form,onSubmit:J},r.default.createElement("label",null,o,r.default.createElement("input",{type:"text",value:b,name:"name",autoFocus:!0,onChange:c=>{L(c.target.value)},placeholder:"Tom Jerry"})),r.default.createElement("label",null,i,r.default.createElement("input",{type:"email",value:y,name:"email",onChange:c=>{P(c.target.value)},placeholder:"mail@mail.com"})),r.default.createElement("label",{className:l.textareaLabel},e,r.default.createElement("textarea",{value:x,placeholder:T,name:"verbatim",rows:3,onChange:c=>{V(c.target.value)}})),r.default.createElement("div",{className:l.formFooter},r.default.createElement("div",{className:l.formActionsContainer},r.default.createElement("button",{type:"submit",disabled:!q||Z},n),r.default.createElement("button",{className:l.closeButton,onClick:D,type:"button"},t)),!U&&r.default.createElement("div",{className:l.ad},r.default.createElement("a",{href:"https://highlight.io",target:"_blank"},r.default.createElement("div",{className:l.logoContainer},r.default.createElement("img",{src:"https://www.highlight.io/images/logo-on-dark.png",alt:"Highlight",className:l.logo}),r.default.createElement("span",null,"Powered by highlight.io"))))))))))}var j={componentStack:null,error:null,showingDialog:!1},E=class extends u.default.Component{constructor(){super(...arguments);this.state=j;this.resetErrorBoundary=()=>{let{onReset:o}=this.props,{error:i,componentStack:n}=this.state;o&&o(i,n),this.setState(j)};this.hideDialog=()=>{this.setState(h(g({},this.state),{showingDialog:!1})),(this.props.onAfterReportDialogCancelHandler||(()=>{window.location.href=window.location.origin}))()};this.onReportDialogSubmitHandler=()=>{this.props.onAfterReportDialogSubmitHandler&&this.props.onAfterReportDialogSubmitHandler()}}componentDidCatch(o,i){var f,d,_;let{beforeCapture:n,onError:a,showDialog:s}=this.props;n&&n(o,(f=i.componentStack)!=null?f:null),so(o,i),a&&a(o,(d=i.componentStack)!=null?d:""),s!==!1&&this.setState(h(g({},this.state),{showingDialog:!0})),this.setState({error:o,componentStack:(_=i.componentStack)!=null?_:null})}componentDidMount(){let{onMount:o}=this.props;o&&o()}componentWillUnmount(){let{error:o,componentStack:i}=this.state,{onUnmount:n}=this.props;n&&n(o,i)}render(){let{fallback:o,children:i,customDialog:n}=this.props,{error:a,componentStack:s,showingDialog:f}=this.state;if(a){let d;return typeof o=="function"?d=o({error:a,componentStack:s,resetError:this.resetErrorBoundary}):d=o,u.default.isValidElement(d)?u.default.createElement(u.default.Fragment,null,f&&u.default.createElement(S,h(g({},this.props.dialogOptions),{onCloseHandler:this.hideDialog,onSubmitHandler:this.onReportDialogSubmitHandler})),d):(o&&console.warn("fallback did not produce a valid ReactElement"),f&&n?n:f&&u.default.createElement(S,h(g({},this.props.dialogOptions),{onCloseHandler:this.hideDialog,onSubmitHandler:this.onReportDialogSubmitHandler})))}return typeof i=="function"?i():i}};function so(t,e){var i;let o=co((i=e.componentStack)!=null?i:"");window.H?(console.error("Highlight ErrorBoundary caught an exception while rendering React component",{error:t}),window.H.consume(t,{payload:{component:o},source:o,type:"React.ErrorBoundary"})):console.warn("You need to install highlight.run.")}function co(t){let e=t.split(`
`);if(e.length<1)return;let i=e[1].trim().split(" ");if(i.length===4)return`<${i[1]}>`}var C=v(require("react")),fo=({children:t})=>{let[e,o]=C.default.useState(!1);if(e)throw new Error("something bad happened - this is a sample test error");return C.default.createElement("button",{type:"button",className:"buggyButton",onClick:()=>o(!0)},t!=null?t:"Throw an Error")};0&&(module.exports={ErrorBoundary,ReportDialog,SampleBuggyButton});
/*! Bundled license information:

classnames/index.js:
  (*!
  	Copyright (c) 2018 Jed Watson.
  	Licensed under the MIT License (MIT), see
  	http://jedwatson.github.io/classnames
  *)
*/
